### PR TITLE
Lint staged

### PR DESCRIPTION
**Pull request by maryamtahhan**
### What does this PR do?
Enable lint staged in husky. depends on https://github.com/containers/podman-desktop-extension-ai-lab/pull/1984

### How to test this PR?

Make a change in a markdown file, commit the change to git and see that the appropriate checks are invoked

```
❯ git commit --signoff -m "chore: test"
✔ Preparing lint-staged...
❯ Running tasks for staged files...
  ❯ package.json — 1 file
    ❯ *.{js,ts,tsx} — 1 file
      ⠙ eslint . --cache --fix
      ◼ prettier --write -u
    ↓ *.{json,md} — no files
◼ Applying modifications from tasks...
◼ Cleaning up temporary files...
```

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+pnpm dlx commitlint --edit $1
```
Comment (jeffmaury) [inline .husky/commit-msg:18]: Please copy this file from https://github.com/containers/podman-desktop-extension-bootc/blob/main/.husky/commit-msg

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+pnpm dlx lint-staged
```
Comment (jeffmaury) [inline .husky/pre-commit:18]: Please copy this file from https://github.com/containers/podman-desktop-extension-bootc/blob/main/.husky/pre-commit

File: package.json [hunk 3]
```diff
@@ -27,13 +27,23 @@
     "typecheck:shared": "tsc --noEmit --project packages/shared",
     "typecheck:frontend": "tsc --noEmit --project packages/frontend",
     "typecheck:backend": "cd packages/backend && npm run typecheck",
-    "typecheck": "npm run typecheck:shared && npm run typecheck:frontend && npm run typecheck:backend"
+    "typecheck": "npm run typecheck:shared && npm run typecheck:frontend && npm run typecheck:backend",
+    "prepare": "husky && husky install"
```
Comment (jeffmaury) [inline package.json:31]: husky install is not required anymore

File: package.json [hunk 4]
```diff
@@ -60,7 +70,8 @@
     "typescript": "5.6.3",
     "typescript-eslint": "^8.11.0",
     "vite": "^5.4.9",
-    "vitest": "^2.0.5"
+    "vitest": "^2.0.5",
+    "husky": "^8.0.0"
```
Comment (jeffmaury) [inline package.json:74]: Use latest version of husky

**Review by jeffmaury — CHANGES_REQUESTED** [review]

**Comment by maryamtahhan** [comment]
Putting all changes into https://github.com/containers/podman-desktop-extension-ai-lab/pull/1984

**Pull request closed without merging**